staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go

```diff
@@ -163,7 +163,40 @@ func (g *Cloud) InstanceShutdownByProviderID(ctx context.Context, providerID str
 
 // InstanceMetadataByProviderID returns metadata of the specified instance.
 func (g *Cloud) InstanceMetadataByProviderID(ctx context.Context, providerID string) (*cloudprovider.InstanceMetadata, error) {
-	return nil, fmt.Errorf("unimplemented")
+	md := &cloudprovider.InstanceMetadata{}
+	var err error
+
+	if providerID == "" {
+		return nil, fmt.Errorf("no providerID")
+	}
+
+	md.ProviderID = providerID
+	timeoutCtx, cancel := context.WithTimeout(ctx, 1*time.Hour)
```

Comment:
nit: let's move it to the top of the function (consistency with the rest of this file)